### PR TITLE
Include strings from Jetpack in bundle

### LINKED ISSUE #2615: Strings from Jetpack are not included in the bundle
**Describe the bug**
When we run our bundle script we are not including strings from jetpack in the resulting bundle.

**To Reproduce**
1. `npm run genstrings`
2. Observe that the strings files (i.e., `bundle/android/strings.xml`) do not contain relevant strings from the jetpack repo: for example, ["Link address to Google Maps"](https://github.com/Automattic/jetpack/blob/f6f313d268c336491c8bca1afcf3e5d3663597ce/extensions/blocks/contact-info/address/edit.native.js#L42)). 

**Expected behavior**
Relevant strings from the jetpack repos should be included in our bundles for translation.

### PR DESCRIPTION
**Pull request by mchowning**
Fixes #2615 

Updating our bundle scripts to ensure that the strings from the jetpack/ repo are included. We are first retrieving the strings from the jetpack repo in a `jetpack-$PLATFORM.pot` file. I am then merging that into the `gutenberg-$PLATFORM.pot` file along with the strings from gutenberg. That `gutenberg-$PLATFORM.pot` file is then passed to the relevant `./bin/po2XXXX.js` script.

Note that I created this branch off of `trunk` in order to avoid having additional string changes on account of the updates to the submodules on `develop`. I anticipate that before merging this PR I'll want to rebase this PR to move the package.json changes on top of the appropriate branch (probably `develop`) and drop the strings changes I have committed in 68a00fd.

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/docs/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,15 +66,14 @@
...
     "prewp": "npm run install:wpcli",
     "wp": "php -d memory_limit=4G bin/wp-cli.phar",
-    "makepot:android": "npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.android.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain gutenberg-android.pot",
-    "makepot:ios": "npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.ios.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain gutenberg-ios.pot",
+    "makepot": "npm run wp -- i18n make-pot ./jetpack/extensions --include=*.native.js,*.$PLATFORM.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain jetpack-$PLATFORM.pot && npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.$PLATFORM.js --exclude=test/* --subtract=./gutenberg.pot --merge=./jetpack-$PLATFORM.pot --ignore-domain gutenberg-$PLATFORM.pot ",
     "premakepot:gutenberg": "npm run clean:gutenberg && npm run build:gutenberg",
     "makepot:gutenberg": "npm run wp -- i18n make-pot ./gutenberg/build --ignore-domain gutenberg.pot",
     "postmakepot:gutenberg": "npm run clean:gutenberg",
     "pregenstrings": "test -f gutenberg.pot || npm run makepot:gutenberg",
```
Comment (mchowning) [inline package.json:73]: I am not sure if there's anything similar we need to do for Jetpack like we need to do for gutenberg. The Jetpack submodule we have checked out [has a `languages/` folder](https://github.com/Automattic/jetpack/tree/f6f313d268c336491c8bca1afcf3e5d3663597ce/languages), that includes, among other things, `jetpack.pot` file. That `jetpack.pot` file does not, however, appear to have all the strings we need since it is missing the ["Link address to Google Maps"](https://github.com/Automattic/jetpack/blob/f6f313d268c336491c8bca1afcf3e5d3663597ce/extensions/blocks/contact-info/address/edit.native.js#L42) string (I'm guessing because it is in a `native.js` file). In addition, that entire `languages/` folder [has been removed on `master`](https://github.com/Automattic/jetpack/pull/16518). 

Basically, there is more work to be done here to get a better understanding of the jetpack build process and either (a) verify that we do not need to do anything more to get all the relevant jetpack strings; or (b) run any necessary jetpack builds as part of the bundling process.

File: package.json [hunk 2]
```diff
@@ -66,15 +66,14 @@
     "install:wpcli": "(test -x bin/wp-cli.phar || curl -Ls https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar -o bin/wp-cli.phar && chmod +x bin/wp-cli.phar) && bin/wp-cli.phar --info",
     "prewp": "npm run install:wpcli",
     "wp": "php -d memory_limit=4G bin/wp-cli.phar",
-    "makepot:android": "npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.android.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain gutenberg-android.pot",
-    "makepot:ios": "npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.ios.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain gutenberg-ios.pot",
+    "makepot": "npm run wp -- i18n make-pot ./jetpack/extensions --include=*.native.js,*.$PLATFORM.js --exclude=test/* --subtract=./gutenberg.pot --ignore-domain jetpack-$PLATFORM.pot && npm run wp -- i18n make-pot ./gutenberg/packages --include=*.native.js,*.$PLATFORM.js --exclude=test/* --subtract=./gutenberg.pot --merge=./jetpack-$PLATFORM.pot --ignore-domain gutenberg-$PLATFORM.pot ",
```
Comment (Tug) [inline package.json:69]: Do you think we could reduce the amount of pot files generated by having something like:
```
i18n make-pot . --include=./jetpack/extensions/**/*.native.js,./jetpack/extensions/**/*.$PLATFORM.js,./gutenberg/packages/**/*.native.js,./gutenberg/packages/**/*.$PLATFORM.js
```
?

**Review by Tug — COMMENTED** [review]
Code looks good. One problem with this approach though is that, as you mentioned, we don't (yet) include the strings from the jetpack project, meaning that we won't have all the strings from jetpack (only those from .native.js and .platform.js files).
Moreover, including all the strings from jetpack might be overkill as it's a much larger project which has many strings.

I tried a different approach [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2622) though it has its drawbacks (the fact that we lose translator comments)

**Comment by mchowning** [comment]
Thanks for reviewing @Tug!

> Do you think we could reduce the amount of pot files generated by having something like

Makes sense. I can try that.

> we won't have all the strings from jetpack (only those from .native.js and .platform.js files).

I see what you mean. Definitely need to fix that if we use this approach.

> including all the strings from jetpack might be overkill as it's a much larger project which has many strings.

I agree, this seems like a pretty big concern.

> I tried a different approach here though it has its drawbacks (the fact that we lose translator comments)

I like the approach in your PR better. It seems much more robust. Like you said, it does seem like we need to figure out the translator's comments issue first. I'm inclined to put more work toward fixing up your approach over the approach in this PR.

**Comment by cameronvoell** [comment]
@mchowning Just reminding that we'll be cutting the 1.38 release branch later in the day Monday 9/28. Please try to get this merged by then or bump the milestone. Thanks much 🙇

**Comment by mchowning** [comment]
Going ahead and closing this since our translations pipeline has been updated to address this in a different way.
